R/poisson.R

```diff
@@ -142,3 +142,29 @@ cdf.poisson <- function(d, x, ...) {
 quantile.poisson <- function(d, p, ...) {
   qpois(p = p, lambda = d$lambda)
 }
+
+#' Fit an poisson distribution to data
+#'
+#' @param d An `poisson` object created by a call to [poisson()].
+#' @param x A vector to fit the distribution to.
+#'
+#' @family poisson distribution
+#'
+#' @return An `poisson` object.
+#' @export
+fit_mle.poisson <- function(d, x, ...) {
+  ss <- suff_stat(d, x, ...)
+  poisson(ss$sum / ss$samples)
```

Comment:
Let's start a `conventions` vignette to keep track of all these names for things to make sure to be consistent down the line. We didn't do that in `broom` and it bit us in the ass.

Reply:
Yeah, already some divergence on my part. `suff_stat.binomial` uses the `trials` rather than `size`. I was aware that it didn't match the `size` argument, but it just didn't seem informative or clear enough in that context. I'm happy to follow whatever convention though of course.

Reply:
I started a list in https://github.com/alexpghayes/distributions/issues/4 that we can use to track for the time being